services/visual_recognition/v3.js

```diff
@@ -81,49 +70,67 @@ module.exports = function (RED) {
     });
   };
 
-  // Utility function that performs the Watson Visual Recognition call. 
-  // the cleanup removes the temp storage, and I am not sure whether 
-  // it should be called here or after alchemy returns and passed
-  // control back to cbdone.
+  // This function performs the operation to Delete ALL Dialogs
+  function performDeleteAllClassifiers(visualRecognition, node, msg) {
 
-  function performAction(params, feature, cbdone) {
-    var visualRecognition = watson.visual_recognition({
-      api_key: apikey,
-      version: 'v3',
-      version_date: '2016-05-19'
-    });
+    node.status({fill:'blue', shape:'dot', text:'requesting Delete All classifiers'});
+
+    var params = {};
+    visualRecognition.listClassifiers(params, function(err, body) {
+      node.status({});
+      if (err) {
+          node.status({fill:'red', shape:'ring', 
+            text:'Delete All : call to listClassifiers failed'});
+          node.error(err, msg);   
+        } else {
+          // Array to hold async tasks
+          var asyncTasks = [];
+          var nbTodelete = body.classifiers.length;
+          var nbdeleted = 0;
+          body.classifiers.forEach(function (aClassifier) {
+              asyncTasks.push(function (cb) {
+                var parms = {};
+                parms.classifier_id=aClassifier.classifier_id;
+                visualRecognition.deleteClassifier(parms, function(err, body) {
+                if (err) {
+                    node.error(err, msg);
+                    console.log('Error with the removal of classifier_id '
+                      +parms.classifier_id +' : ' +  err);
+                    cb('error');
+                  } else {
+                    console.log('Classifier ID '+ aClassifier.classifier_id 
+                      + ' deleted successfully.');
+                    nbdeleted++;
+                  }
+                  cb(null,parms.classifier_id);
+                });  
+              });
+          });
+        } // else
+        async.parallel(asyncTasks, function(error, deletedList){
+          if (deletedList.length===nbTodelete) {
+            msg.payload='see msg.result.error';
+            msg.result = 'All custom classifiers have been deleted.';
+          }
+          else {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Closing curly brace does not appear on the same line as the subsequent block.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2714202781/issues/source?bid=3225115&fileBranchId=3344975#l115)